ui/src/message_popup/author/question_page.jsx

```diff
@@ -150,62 +166,75 @@ export default React.createClass({
   onDeleteQuestion(){
     console.log("");
     console.log("Deleting and archiving the following question:");
-    console.log(this.props.originalQuestion);
+    const {originalQuestion} = this.props;
+    console.log(originalQuestion);
     console.log("");
+    var {currentQuestions, archivedQuestions} = this.props.allQuestions;
+    if(originalQuestion !== undefined){
+      currentQuestions = currentQuestions.filter(questionInner => questionInner.id !== originalQuestion.id);
+      archivedQuestions = archivedQuestions.concat(originalQuestion);
+    }
+    Api.saveQuestions({currentQuestions, archivedQuestions});
+    this.setState({currentQuestions, archivedQuestions});
     this.onReturnToQuestions();
   },
 
   render(){
+    const {loaded, originalQuestion} = this.props;
+    const {questionText, students, goodExamplesText, badExamplesText, indicator, deleteConfirmationOpen} = this.state;
+    const title = originalQuestion !== undefined ? `Editing Question #${originalQuestion.id}` : loaded ? 'New Question' : '';
```

Comment:
Can we remove `loaded` altogether, and have the parent component not render `QuestionPage` until it has loaded the question?